crates/modelardb_common/src/storage.rs

```diff
@@ -0,0 +1,222 @@
+    use tempfile::{self, TempDir};
+
+    use crate::test;
+
+    // Tests for writing and reading Apache Parquet files.
+    #[test]
+    fn test_write_batch_to_apache_parquet_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let batch = test::compressed_segments_record_batch();
+
+        let apache_parquet_path = temp_dir.path().join("test.parquet");
+        write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None).unwrap();
+
+        assert!(apache_parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_empty_batch_to_apache_parquet_file() {
+        let fields: Vec<Field> = vec![];
+        let schema = Schema::new(fields);
+        let batch = RecordBatch::new_empty(Arc::new(schema));
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let apache_parquet_path = temp_dir.path().join("empty.parquet");
+        write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None).unwrap();
+
+        assert!(apache_parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_invalid_extension() {
+        write_to_file_and_assert_failed("test.txt".to_owned());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_no_extension() {
+        write_to_file_and_assert_failed("test".to_owned());
+    }
+
+    fn write_to_file_and_assert_failed(file_name: String) {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let batch = test::compressed_segments_record_batch();
+
+        let apache_parquet_path = temp_dir.path().join(file_name);
+        let result =
+            write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None);
+
+        assert!(result.is_err());
+        assert!(!apache_parquet_path.exists());
+    }
+
+    #[tokio::test]
+    async fn test_read_entire_apache_parquet_file() {
+        let file_name = "test.parquet".to_owned();
+        let (_temp_dir, path, batch) = create_apache_parquet_file_in_temp_dir(file_name);
+
+        let result = read_batch_from_apache_parquet_file(path.as_path()).await;
+
+        assert!(result.is_ok());
+        assert_eq!(batch, result.unwrap());
+    }
+
+    #[tokio::test]
+    async fn test_read_from_non_apache_parquet_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join("test.txt");
+        File::create(path.clone()).unwrap();
+
+        let result = read_batch_from_apache_parquet_file(path.as_path());
+
+        assert!(result.await.is_err());
+    }
+
+    #[tokio::test]
+    async fn test_read_from_non_existent_path() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join("none.parquet");
+        let result = read_batch_from_apache_parquet_file(path.as_path());
+
+        assert!(result.await.is_err());
+    }
+
+    /// Create an Apache Parquet file in the [`tempfile::TempDir`] from a generated [`RecordBatch`].
+    fn create_apache_parquet_file_in_temp_dir(
```

Comment:
Maybe also run the tests that should successfully read `RecordBatch`es with Apache Parquet files that are designed to have multiple `Row Groups` and thus would have failed without the changes in this PR, i. e., specifically adding tests that fails with the old code.

Reply:
Fixed.